rockcraft/oci.py

```diff
@@ -45,29 +47,39 @@ class Image:
 
     @classmethod
     def from_docker_registry(
-        cls, image_name: str, *, image_dir: Path
+        cls, image_name: str, *, image_dir: Path, arch: str, variant: str
     ) -> Tuple["Image", str]:
         """Obtain an image from a docker registry.
 
         :param image_name: The image to retrieve, in ``name:tag`` format.
         :param image_dir: The directory to store local OCI images.
+        :param arch: The architecture of the Docker image to fetch.
+        :param variant: The variant, if any, of the Docker image to fetch.
+
 
         :returns: The downloaded image and it's corresponding source image
         """
         image_dir.mkdir(parents=True, exist_ok=True)
         image_target = image_dir / image_name
 
         source_image = f"docker://{image_name}"
-        _copy_image(source_image, f"oci:{image_target}")
+        platform_params = ["--override-arch", arch]
+        if variant:
+            platform_params += ["--override-variant", variant]
+        _copy_image(source_image, f"oci:{image_target}", *platform_params)
 
         return cls(image_name=image_name, path=image_dir), source_image
 
     @classmethod
-    def new_oci_image(cls, image_name: str, image_dir: Path) -> Tuple["Image", str]:
+    def new_oci_image(
+        cls, image_name: str, image_dir: Path, arch: str, variant: str
+    ) -> Tuple["Image", str]:
         """Create a new OCI image out of thin air.
 
         :param image_name: The image to initiate, in ``name:tag`` format.
         :param image_dir: The directory to store the local OCI image.
+        :param arch: The architecture of the OCI image to create.
+        :param variant: The variant, if any, of the OCI image to create.
```

Comment:
> `The variant, if any,`

So should this be an `Optional[str]`? Might be a bit more ergonomic than having to explicitly pass `""`

Reply:
yep good catch
https://github.com/canonical/rockcraft/pull/52/commits/9811b0d8ced59a6b80dc451e549cb50b2c76f2fe